packages/relay/tests/lib/eth.spec.ts

```diff
@@ -371,6 +378,30 @@ describe('Eth calls using MirrorNode', async function () {
     verifyBlockConstants(result);
   });
 
+  it('eth_getBlockByNumber with zero transactions', async function () {
+    // mirror node request mocks
+    mock.onGet(`blocks/${blockNumber}`).reply(200, defaultBlock);
+    mock.onGet(`contracts/results?timestamp=gte:${defaultBlock.timestamp.from}&timestamp=lte:${defaultBlock.timestamp.to}`).reply(200, defaultContractResults);
```

Comment:
A better way would be modify defaultContractResults to return an empty array here.
In this way the follow up `contracts/${contractAddress1}/results` are not made and the mocking is not needed. 
```suggestion
    mock.onGet(`contracts/results?timestamp=gte:${defaultBlock.timestamp.from}&timestamp=lte:${defaultBlock.timestamp.to}`).reply(200, { 'results': []});
```